pennylane/vqe/vqe.py

```diff
@@ -136,6 +136,8 @@ class VQECost:
             where ``params`` are the trainable weights of the variational circuit, and
             ``kwargs`` are any additional keyword arguments that need to be passed
             to the template.
+        hamiltonian (qml.vqe.Hamiltonian): Hamiltonian operator (represented using the class
+            :class:`~.Hamiltonian`) whose expectation value should be measured.
```

Comment:
It would need to be the full path (i.e., `pennylane.vqe.Hamiltonian`) for Sphinx to resolve it. However, more maintainable to just use `~`, so Sphinx will look for the first match in the code base. Assuming we never have two classes called `Hamiltonian` 😆 
```suggestion
        hamiltonian (~.Hamiltonian): Hamiltonian operator (represented using the class
            :class:`~.Hamiltonian`) whose expectation value should be measured.
```

Reply:
nitpick: We could also remove `(represented using the class :class:`~.Hamiltonian`)`, since Sphinx will make the arg type a hyperlink to the Hamiltonian class, but up to you (don't mind either way)

Reply:
works better, thanks